internal/stack/serverless.go

```diff
@@ -273,10 +281,23 @@ func (sp *serverlessProvider) BootUp(options Options) error {
 		printUserConfig(options.Printer, config)
 	}
 
-	logger.Infof("Starting local agent")
-	err = sp.startLocalAgent(options, config)
+	logger.Infof("Starting local services")
+	err = sp.startLocalServices(options, config)
+	if err != nil {
+		return fmt.Errorf("failed to start local services: %w", err)
+	}
+
+	//Re-create clients if running on an existing project
+	if sp.kibanaClient == nil {
+		if err = sp.createClients(project); err != nil {
+			return err
+		}
+	}
+
+	// Updating the output with ssl certificates created in startLocalServices
+	err = project.UpdateLogstashFleetOutput(sp.profile, sp.kibanaClient)
 	if err != nil {
-		return fmt.Errorf("failed to start local agent: %w", err)
+		return err
 	}
```

Comment:
>It looks like that this code could be moved to the createProject():

I have already added the change -  `Add` happens in `CreateProject` - https://github.com/elastic/elastic-package/pull/1646/files#diff-b4efb1ac558d15d3077525cb9be59dce12ee723a4749253e9ca7cc860419f1f8R109

But at this time ssl certificates for `elastic-agent client` are not created. Hence an `Update` shall be called once the `startLocalServices` is done and certificates are created.

>That function has already the clients created (no need to re-create them), and the services should be available at the end of that function (Fleet API).

Re-creation of clients is because - once the project is created and we want to change the local services for some reason , it is not possible to update the fleet output [ If the certificates are regenerated , for example]. Because the `kibanaClient` is nil  in `BootUp` if there is an [existing project](https://github.com/elastic/elastic-package/pull/1646/files#diff-b4efb1ac558d15d3077525cb9be59dce12ee723a4749253e9ca7cc860419f1f8L247) the clients are not available and there will be a `panic`.

>If it needs to be kept here, since it requires the files created during startLocalServices(), could it be just run this method during the creation phase ? Maybe, that would mean just run that if sp.KibanaClient != nil , since those clients are created if the project is being created. To avoid re-adding the same output in Fleet, or at least repeating the query.

I think we cannot run `startLocalServices` until we have the stack up , because agent probably needs `es`, `fleet server` to be up? But I see what you mean. We do not want to run the `PUT` query if there is no change in the input  you mean? But the problem is we do not know if the input [ SSL certificates in this case ] changed? Do the certificates , once created , change at all? Is there a validity period tied to them?

Reply:
I see, thanks for the clarifications!

> But at this time ssl certificates for elastic-agent client are not created. Hence an Update shall be called once the startLocalServices is done and certificates are created.

Right, resources like those certificates are created when `startLocalServices` is executed, and as it was checked here https://github.com/elastic/elastic-package/pull/1646#discussion_r1480075215 fleet output needs to be updated so the SSL settings are also updated in Fleet.

> I think we cannot run startLocalServices until we have the stack up , because agent probably needs es, fleet server to be up?

Exactly, at least when logstash is not part of the scenario, elastic-agent needs Fleet (and kibana and ES) running and ready.


> We do not want to run the PUT query if there is no change in the input you mean? But the problem is we do not know if the input [ SSL certificates in this case ] changed? Do the certificates , once created , change at all? Is there a validity period tied to them?

As certificates are updated, I think it requires to be run again. Currently, I cannot think how to manage this in other way.